src/server/qgsconfigcache.h

```diff
@@ -30,20 +31,59 @@
 #include "qgsproject.h"
 #include "qgsserversettings.h"
 
+#ifndef SIP_RUN
+
+
+#endif
+
+class QgsConfigCache;
+
+/**
+ * \ingroup server
+ * \brief Abstract cache strategy for server configuration
+ * \since QGIS 3.24
+ */
+class SERVER_EXPORT QgsAbstractCacheStrategy  : public QObject
+{
+    friend class QgsConfigCache;
+
+    Q_OBJECT
+  public:
+    virtual QString ident() const = 0;
```

Comment:
Can this be called `name` o(I think that would be consistent with other naming choices you did in this pr?) Or otherwise a full name like `identifier`or similar.

Reply:
Fixed,

> > Note for the protected virtuals, the requirement was:
> > 
> > * Not public
> > * Overridable
> > * Callable from QgsConfigCache
> 
> Nr 1 and 3 are mutually exclusive I think?
> 
> In general I prefer the Herb Sutter style: Public non virtual that calls private virtual.

That's why I use friendship.